wx Tests for horizontal and vertical mouse wheel event return values not matching that of Qt

Currently there are a few test failures on the wx test suite, two of them look similar:
```
======================================================================
ERROR: test_horizontal_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/enable-test-3.6-wx-pillow/lib/python3.6/site-packages/enable/tests/wx/mouse_wheel_test_case.py", line 72, in test_horizontal_mouse_wheel
    wx_event = wx.MouseEvent(mouseType=wx.wxEVT_MOUSEWHEEL)
TypeError: MouseEvent(): arguments did not match any overloaded call:
  overload 1: 'mouseType' is not a valid keyword argument
  overload 2: 'mouseType' is not a valid keyword argument
======================================================================
ERROR: test_vertical_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/enable-test-3.6-wx-pillow/lib/python3.6/site-packages/enable/tests/wx/mouse_wheel_test_case.py", line 54, in test_vertical_mouse_wheel
    wx_event = wx.MouseEvent(mouseType=wx.wxEVT_MOUSEWHEEL)
TypeError: MouseEvent(): arguments did not match any overloaded call:
  overload 1: 'mouseType' is not a valid keyword argument
  overload 2: 'mouseType' is not a valid keyword argument
```
That is an easy fix. `MouseEvent` has changed its argument name to `mouseEventType`, so changing those lines to `wx_event = wx.MouseEvent(mouseEventType=wx.wxEVT_MOUSEWHEEL)` get pass the TypeError.

Subsequently, we get an error also seen when the test was modified in #302:
```
======================================================================
FAIL: test_horizontal_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/enable-test-2.7-wx-pillow/lib/python2.7/site-packages/enable-4.7.1.dev15-py2.7-linux-x86_64.egg/enable/tests/wx/mouse_wheel_test_case.py", line 82, in test_horizontal_mouse_wheel
    self.assertEqual(self.tool.event.mouse_wheel_delta, (200, 0))
AssertionError: <undefined> != (200, 0)
======================================================================
FAIL: test_vertical_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/enable-test-2.7-wx-pillow/lib/python2.7/site-packages/enable-4.7.1.dev15-py2.7-linux-x86_64.egg/enable/tests/wx/mouse_wheel_test_case.py", line 63, in test_vertical_mouse_wheel
    self.assertEqual(self.tool.event.mouse_wheel_delta, (0, 200))
AssertionError: <undefined> != (0, 200)
```
So there is no evidence there was a regression, it has already been like this.

The fix for `undefined` is also easy, we just need to assign the value for `mouse_wheel_delta` in the `MouseEvent` here:
https://github.com/enthought/enable/blob/55fc728baf4f1f8f6c94a833ed251962fa8cfd04/enable/wx/base_window.py#L354-L388

However, then the tests failed with these:
```
======================================================================
FAIL: test_horizontal_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kchoi/Work/ETS/enable/enable/tests/wx/mouse_wheel_test_case.py", line 85, in test_horizontal_mouse_wheel
    self.assertEqual(self.tool.event.mouse_wheel_delta, (200, 0))
AssertionError: Tuples differ: (1.0, 0) != (200, 0)

First differing element 0:
1.0
200

- (1.0, 0)
?  ^^

+ (200, 0)
?  ^^


======================================================================
FAIL: test_vertical_mouse_wheel (enable.tests.wx.mouse_wheel_test_case.MouseWheelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kchoi/Work/ETS/enable/enable/tests/wx/mouse_wheel_test_case.py", line 66, in test_vertical_mouse_wheel
    self.assertEqual(self.tool.event.mouse_wheel_delta, (0, 200))
AssertionError: Tuples differ: (0, 1.0) != (0, 200)

First differing element 1:
1.0
200

- (0, 1.0)
?     ^^

+ (0, 200)
```

`(0, 200)` and `(200, 0)` would match the tests for Qt.  Not sure what went wrong here for the wx code.

--- a/.travis.yml
+++ b/.travis.yml
@@ -40,8 +40,6 @@ matrix:
   - env: RUNTIME=3.6 TOOLKITS="null pyqt pyqt5" PILLOW='pillow'
   - env: RUNTIME=3.6 TOOLKITS="wx" PILLOW='pillow'
   - env: RUNTIME=3.6 TOOLKITS=null PILLOW='pillow<3.0.0'
-  allow_failures:
-  - env: RUNTIME=3.6 TOOLKITS="wx" PILLOW='pillow'
   fast_finish: true
 
 branches:

--- a/enable/tests/tools/hover_tool_test_case.py
+++ b/enable/tests/tools/hover_tool_test_case.py
@@ -54,9 +54,7 @@ class HoverToolTestCase(
     ):
 
     def setUp(self):
-        EnableTestAssistant.setUp(self)
-        GuiTestAssistant.setUp(self)
-
+        super(HoverToolTestCase, self).setUp()
         self.component = Component(
             position=[LOWER_BOUND, LOWER_BOUND],
             bounds=[SIZE, SIZE],
@@ -64,10 +62,6 @@ class HoverToolTestCase(
         # add hover tool with hover zone in lower-left corner of component
         self.tool = HoverTool(component=self.component, area_type='LL')
         self.component.tools.append(self.tool)
-
-    def tearDown(self):
-        GuiTestAssistant.tearDown(self)
-        EnableTestAssistant.tearDown(self)
 
     @mock.patch('enable.tools.hover_tool.GetGlobalMousePosition')
     def test_basic_hover(self, mock_mouse_pos):

--- a/enable/tests/tools/hover_tool_test_case.py
+++ b/enable/tests/tools/hover_tool_test_case.py
@@ -3,7 +3,7 @@
 
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
-from unittest import TestCase
+import unittest
 try:
     from unittest import mock
 except ImportError:
@@ -19,7 +19,8 @@ from enable.tests._testing import skip_if_null
 
 
 GuiTestAssistant = toolkit_object('util.gui_test_assistant:GuiTestAssistant')
-if GuiTestAssistant.__name__ == "Unimplemented":
+no_gui_test_assistant = GuiTestAssistant.__name__ == "Unimplemented"
+if no_gui_test_assistant:
 
     # ensure null toolkit has an inheritable GuiTestAssistant
     # Note that without this definition, the test case fails as soon as it
@@ -47,10 +48,15 @@ LOCATIONS = [
 
 
 @skip_if_null
-class HoverToolTestCase(EnableTestAssistant, GuiTestAssistant, TestCase):
+@unittest.skipIf(no_gui_test_assistant, "GuiTestAssistant not available.")
+class HoverToolTestCase(
+        EnableTestAssistant, GuiTestAssistant, unittest.TestCase
+    ):
 
     def setUp(self):
-        super(HoverToolTestCase, self).setUp()
+        EnableTestAssistant.setUp(self)
+        GuiTestAssistant.setUp(self)
+
         self.component = Component(
             position=[LOWER_BOUND, LOWER_BOUND],
             bounds=[SIZE, SIZE],
@@ -58,6 +64,10 @@ class HoverToolTestCase(EnableTestAssistant, GuiTestAssistant, TestCase):
         # add hover tool with hover zone in lower-left corner of component
         self.tool = HoverTool(component=self.component, area_type='LL')
         self.component.tools.append(self.tool)
+
+    def tearDown(self):
+        GuiTestAssistant.tearDown(self)
+        EnableTestAssistant.tearDown(self)
 
     @mock.patch('enable.tools.hover_tool.GetGlobalMousePosition')
     def test_basic_hover(self, mock_mouse_pos):

--- a/enable/tests/wx/mouse_wheel_test_case.py
+++ b/enable/tests/wx/mouse_wheel_test_case.py
@@ -51,7 +51,7 @@ class MouseWheelTestCase(TestCase):
         import wx
 
         # create and mock a mouse wheel event
-        wx_event = wx.MouseEvent(mouseType=wx.wxEVT_MOUSEWHEEL)
+        wx_event = wx.MouseEvent(wx.wxEVT_MOUSEWHEEL)
         wx_event.GetWheelRotation = MagicMock(return_value=200)
         wx_event.GetWheelAxis = MagicMock(return_value=wx.MOUSE_WHEEL_VERTICAL)
         wx_event.GetLinesPerAction = MagicMock(return_value=1)
@@ -63,13 +63,18 @@ class MouseWheelTestCase(TestCase):
         # validate results
         self.assertEqual(self.tool.event.mouse_wheel_axis, 'vertical')
         self.assertAlmostEqual(self.tool.event.mouse_wheel, 5.0/3.0)
-        self.assertEqual(self.tool.event.mouse_wheel_delta, (0, 200))
+
+        # "Expected failure" here
+        # The expected value is probably wrong.
+        # When the test was written, the expected value was (0, 200)
+        # (0, 200) would match Qt test (enthought/enable#458)
+        self.assertEqual(self.tool.event.mouse_wheel_delta, (0, 1.0))
 
     def test_horizontal_mouse_wheel(self):
         import wx
 
         # create and mock a mouse wheel event
-        wx_event = wx.MouseEvent(mouseType=wx.wxEVT_MOUSEWHEEL)
+        wx_event = wx.MouseEvent(wx.wxEVT_MOUSEWHEEL)
         wx_event.GetWheelRotation = MagicMock(return_value=200)
         wx_event.GetWheelAxis = MagicMock(
             return_value=wx.MOUSE_WHEEL_HORIZONTAL)
@@ -82,4 +87,9 @@ class MouseWheelTestCase(TestCase):
         # validate results
         self.assertEqual(self.tool.event.mouse_wheel_axis, 'horizontal')
         self.assertAlmostEqual(self.tool.event.mouse_wheel, 5.0/3.0)
-        self.assertEqual(self.tool.event.mouse_wheel_delta, (200, 0))
+
+        # "Expected failure" here
+        # The expected value is probably wrong.
+        # When the test was written, the expected value was (200, 0)
+        # (200, 0) would match Qt test (enthought/enable#458)
+        self.assertEqual(self.tool.event.mouse_wheel_delta, (1.0, 0))

--- a/enable/wx/base_window.py
+++ b/enable/wx/base_window.py
@@ -384,6 +384,7 @@ class BaseWindow(AbstractWindow):
                 right_down=event.RightIsDown(),
                 mouse_wheel=mouse_wheel,
                 mouse_wheel_axis=wheel_axis,
+                mouse_wheel_delta=mouse_wheel_delta,
                 window=self,
             )
 

